elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreResourceDefinition.java

```diff
@@ -346,8 +386,76 @@ protected void performRuntime(ModelNode result, OperationContext context, ModelN
+                    break;
 
+                case ElytronDescriptionConstants.READ_ALIASES:
+                    try {
+                        result.set(credentialStoreService.getValue().getAliases().stream().map(ModelNode::new).collect(Collectors.toList()));
+                    } catch (CredentialStoreException e) {
+                        throw new OperationFailedException(e);
+                    }
                     break;
+
+                case ElytronDescriptionConstants.ADD_ALIAS:
+                    try {
+                        String alias = ALIAS.resolveModelAttribute(context, operation).asString();
+                        String entryType = asStringIfDefined(context, ENTRY_TYPE, operation);
+                        String secretValue = asStringIfDefined(context, SECRET_VALUE, operation);
+                        CredentialStore credentialStore = credentialStoreService.getValue();
+                        if (entryType == null || entryType.equals(PasswordCredential.class.getCanonicalName())) {
+                            if (credentialStore.exists(alias, PasswordCredential.class)) {
+                                throw ROOT_LOGGER.credentialAlreadyExists(alias, PasswordCredential.class.getName());
+                            }
+                            storeSecret(credentialStore, alias, secretValue);
+                        } else {
+                            String credentialStoreName = CredentialStoreResourceDefinition.credentialStoreName(operation);
+                            throw ROOT_LOGGER.credentialStoreEntryTypeNotSupported(credentialStoreName, entryType);
+                        }
+                    } catch (CredentialStoreException e) {
+                        throw ROOT_LOGGER.unableToCompleteOperation(e, e.getLocalizedMessage());
+                    }
+                    break;
+
+                case ElytronDescriptionConstants.REMOVE_ALIAS:
+                    try {
+                        String alias = ALIAS.resolveModelAttribute(context, operation).asString();
+                        CredentialStore credentialStore = credentialStoreService.getValue();
+                        PasswordCredential retrieved = credentialStore.retrieve(alias, PasswordCredential.class);
+                        credentialStore.remove(alias, PasswordCredential.class);
+                        try {
+                            credentialStore.flush();
+                        } catch (CredentialStoreException e) {
+                            // the operation fails, return removed entry back to the store to avoid an inconsistency
+                            // between the store on the FS and in the memory
+                            credentialStore.store(alias, retrieved);
+                            throw e;
+                        }
+                    } catch (CredentialStoreException e) {
+                        throw ROOT_LOGGER.unableToCompleteOperation(e, e.getLocalizedMessage());
```

Comment:
This should throw a RuntimeException, not OFE. OFE should only be used for user errors, not server faults.